docs/docs/policies.mdx

```diff
@@ -346,8 +346,56 @@ policies:
 * `enable_fallback_prediction` (default: `true`): Please see the
    [fallback documentation](fallback-handoff.mdx#handling-low-action-confidence) for
    further information.
-* `check_for_contradictions` (default: `true`): After training, the RulePolicy will
-   perform a check to make sure that there are no rules that contradict with each other, or with any stories.
+* `check_for_contradictions` (default: `true`):
+   Before training, the RulePolicy will perform a check to make sure that
+   actions always set slots and active loops that they set in other rules.
+   The following snippet is an example for an incomplete rule:
+
+   ```yaml-rasa
+   rules:
+   - rule: complete rule
+     steps:
+     - intent: do_something
+     - action: action_that_sets_some_slot
+     - slot_was_set:
+       - some_slot: "some value"
+
+   - rule: incomplete rule
+     steps:
+     - intent: do_something
+     - action: action_that_sets_some_slot
+   ```
+
+   In the second `incomplete rule`, `action_that_sets_some_slot` is expected to set
+   `some_slot` but does not do it.
+   There is several possible ways to fix this rule, for example:
+
+   - if `some_slot` should not be set, you should explicitly set it to `null`
+
+   ```yaml-rasa
+   rules:
+   - rule: fixes incomplete rule
+     steps:
+     - intent: do_something
+     - action: action_that_sets_some_slot
+     - slot_was_set:
+       - some_slot: null
+   ```
+
+   - if you want to deal with setting slot in a different rule or story,
```

Comment:
Even though it would solve the contradiction, it seems like bad practice for users to put slot sets only in certain rules and use the `wait_for_user_input` key. I'm leaning towards removing this recommendation.

Reply:
to me it's how rules should be written (especially taken into account that we restrict them by one user turn), but I can remove that

Reply:
also, this is not a recommendation, it is a showcase of possible solutions

Reply:
It solves the problem, but I can see users interpreting this as "use `wait_for_user_input` to solve conflict errors" when they should be re-examining their rule & story definitions.

Reply:
but this is not in the part of the docs where we recommend staff, it is in `policies.mdx` which describe capabilities of the policy. I think users should know that it is possible.
also, I don't see a problem in  "use `wait_for_user_input` to solve conflict errors", if it indeed solves them

Reply:
@erohmensing what do you think?